### DISCUSSION TITLE
how to use git in terminal

### DISCUSSION THREAD

**Discussion by MahaUraga** [community/community#134628] [New to GitHub]
### Select Topic Area

Question

### Body

how to use git in terminal

**Comment by enesaktas10** [comment]
Response#1

**Comment by elsycharles** [comment]
Hi @aminedev0x01 !

1. for using git in terminal first you have to install git
for linux users tap in terminal : sudo apt-get install git -y
for windows user : download it from [https://gitforwindows.org/](https://gitforwindows.org/) and install it

2. there is a lot of command you should know about that but the common git commands are : 
git clone, git pull, git push origin, git add, git commit -m, git merge
check all commands here [https://docs.gitlab.com/ee/gitlab-basics/start-using-git.html](https://docs.gitlab.com/ee/gitlab-basics/start-using-git.html)

Hope that'll help you.

**Comment by swaraj-das** [comment]
# Basic Guide to Using Git in the Terminal

## Setting Up Git

### Install Git
- **Windows**: Download the installer from [git-scm.com](https://git-scm.com/) and follow the instructions.
- **macOS**: Install Git using Homebrew with `brew install git` or Xcode Command Line Tools with `xcode-select --install`.
- **Linux**: Install Git using your package manager, for example, `sudo apt-get install git` on Debian-based systems.

### Configure Git
Set your username and email. These details will be associated with your commits.
```sh
git config --global user.name "Your Name"
git config --global user.email "your.email@example.com"
```
## Basic Git Commands

This guide provides an introduction to commonly used Git commands for version control.

### Getting Started

**Initialize a Git Repository:**

Start by navigating to your project directory in your terminal. Then, initialize a new Git repository using the following command:

```sh git init```

**Clone an Existing Repository:**

To clone an existing repository from a remote server like GitHub, use:

```sh git clone https://github.com/username/repository.git```


Replace `username/repository.git` with the actual URL of the repository you want to clone.

### Managing Changes

**Check the Status of Your Repository:**

Use this command to see the current status of your working directory, including untracked, modified, and staged files:

`git status`

**Add Files to Staging Area:**

- To stage all changes:

`git add .`

- To stage specific files:

`git add file1 file2`

**Commit Changes:**

Commit your staged changes to the repository with a descriptive message using:

`git commit -m "Your commit message"`

**View Commit History:**

This command shows the history of commits made in your repository:

`git log`

### Collaboration (Pushing and Pulling)

**Push Changes to Remote Repository:**

Push your local commits to a remote repository (e.g., GitHub) using:

`git push origin branch-name`

Replace `origin` with the remote name (usually `origin`) and `branch-name` with the actual branch you want to push.

**Pull Changes from Remote Repository:**

Fetch and merge changes from the remote repository into your local branch:
`git pull origin branch-name`

Replace `origin` and `branch-name` as before.

### Branching

**Create a New Branch:**

Create a new branch and switch to it simultaneously:

`git checkout -b new-branch`

**Switch Between Branches:**

Switch to an existing branch:

`git checkout branch-name`

**Merge Branches:**

Merge a branch (e.g., `feature-branch`) into your current branch:

`git merge feature-branch`

**Delete a Branch:**

- To delete a branch locally:

`
git branch -d branch-name`

- To delete a branch remotely:

```
git push origin --delete branch-name
```

**Additional Commands:**

- **Stash Changes:** Temporarily store uncommitted changes:

```
git stash
```

Apply the stashed changes later with:

```
git stash apply
```

- **Rebase Branches:** Apply your local commits on top of another branch:


`git rebase branch-name`


- **View Differences:** See the changes made relative to the last commit:


`git diff`


These commands provide a solid foundation for working with Git. As you gain experience, explore more advanced features and workflows to further streamline your development process.

**Reply by ebndev** [reply]
Hi @swaraj-das, thanks for participating in GitHub Discussions!

Please note: we’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  

You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. 

Please feel encouraged to continue engaging in Discussions as your authentic self though. We appreciate you being a part of the Community 🌟

**Comment by ebndev** [comment]
Hi @aminedev0x01, thanks for posting in GitHub Discussions! 

I've gone ahead and moved this post to our `New to GitHub` category, as this topic may be more relevant there and might help someone give you a nudge in the right direction. 😁

In the meantime, please feel encouraged to explore existing discussions in the category and utilize the search in case other users may have run into the same issue!

**Comment by jessdebrito** [comment]
Hi, @aminedev0x01!
You can install the pack **https://git-scm.com/downloads** 
Remember to use the **BASH** to use the comands, and check what code you need here: https://git-scm.com/docs/git

If will be your first time using the terminal, you need to link with your GitHub account. Exemple:
- `git config --global user.name "John Doe"`
- `git config --global user.email johndoe@example.com`

Important codes are those that allow you to navigate through your computer's folders using the terminal
- See where your terminal is:
`pwd`
- Change the folder
`cd`
Exemple:
`cd D:/Github/Project_1`

- See files and folders(does not include hidden ones):
`ls`
- See files and folders(including hidden ones):
`ls -a`
- Create new repository:
`git init .`

For beginners I recommend to create the repository using the site, before use this basic command to clone: 
`git clone https://github.com/your_repository
`
- Commit
`git commit -m "mensagem"`

To start, these are the main codes. Once you get used to them, I recommend looking at the documentation for the code of **stage**/**unstage** and **Branch**.
